src/common/helpers/utils.ts

```diff
@@ -94,6 +94,39 @@ export function sortRelationshipOrdinal(sortByProperty: string) {
 	};
 }
 
+/**
+ * This function repalces other space control character to U+0020 and trim extra spaces
+ * @param {string} text - text to sanitize
+ * @returns {string} - sanitized text
+ */
+export function collapseWhiteSpaces(text:string):string {
+	const spaceRegex = RegExp(/\s+/gi);
+	const doubleRegex = RegExp(/\s{2,}/gi);
+	let sanitizedText = text.replace(spaceRegex, '\u0020');
+	sanitizedText = text.replace(doubleRegex, '\u0020');
+	return sanitizedText.trim();
+}
+
+/**
+ * This function is to sanitize text inputs
+ * @param {string} text - text to sanitize
+ * @returns {string} - sanitized text
+ */
+export function sanitize(text:string):string {
+	if (!text) {
```

Comment:
Juuuuust in case, since we call `.normalize()`  next we should ensure `text` is indeed a string, otherwise this could break.
lodash has the `isString` utility just for that:

```suggestion
	if (!isString(text)) {
```